File: svg.js
export default {
  template: `
    <svg :viewBox="viewBox" :width="width" :height="height" xmlns="http://www.w3.org/2000/svg">
      <slot></slot>
    </svg>
  `,
  props: {
    width: { type: Number, default: 200 },
    height: { type: Number, default: 200 },
    viewBox: { type: String, default: "0 0 200 200" },
  },
};

File: svg.py


from nicegui.element import Element


class SVG(Element, component="svg.js"):
    def __init__(self, width: int = 200, height: int = 200) -> None:
        super().__init__()
        self.width = width
        self.height = height
        self.children = []

        self._props['width'] = width
        self._props['height'] = height
    
    def add_child(self, child):
        self.children.append(child)
    
    def add_children(self, children):
        self.children.extend(children)

